### ISSUE TITLE
Specmatic not overriding global security

### ISSUE THREAD

**Issue by Rajaneeshkumar** [specmatic/specmatic#1491]
**Description**
In OpenAPI 3, defined a SecurityScheme at global level but then override it at certain endpoints to not use security,but specmatic is not overriding security schema.

**Steps to reproduce**

openapi: 3.0.0
...
# 1) Define the security scheme type (HTTP bearer)
```
components:
  securitySchemes:
    bearerAuth:            # arbitrary name for the security scheme
      type: http
      scheme: bearer
      bearerFormat: JWT    # optional, arbitrary value for documentation purposes
```
# 2) Apply the security globally to all operations
```
security:
  - bearerAuth: []         # use the same name as above
```

# 3) Now I want to override global security schema bearerAuth for specific endpoint

```
paths:
  /unprotected/path:
    get:
      security: []
```
      

**Expected behavior**
Specmatic should not expect bearerAuth for /unprotected/path endpoint but it is throwing below error
**REQUEST\n  \n     Authorization header is missing in request**

**System Information:**
 - OS & version: MAC OS 14.4.1
 - Specmatic version: latest
 - JDK version: openjdk 17.0.11

**Additional context**
This is really blocking me to proceed further , Quick workaround/help would be appreciated , Thanks !!

@harikrishnan83 @samyakOO7 [pranavgawri](https://github.com/pranavgawri)

**Comment by harikrishnan83** [comment]
hello @Rajaneeshkumar, thanks for reporting the issue. We will take a look.

**Comment by Rajaneeshkumar** [comment]
> hello @Rajaneeshkumar, thanks for reporting the issue. We will take a look.

Any update pls !!

**Comment by harikrishnan83** [comment]
hello @Rajaneeshkumar , this issue is fixed and will be part of our upcoming release. I will keep you posted. Thanks.

**Comment by Rajaneeshkumar** [comment]
> hello @Rajaneeshkumar , this issue is fixed and will be part of our upcoming release. I will keep you posted. Thanks.

Thanks !!
Please make this fix available for specmatic-node as well.

**Comment by Rajaneeshkumar** [comment]
@harikrishnan83 When can I expect the upcoming release? waiting for this change !!

**Comment by harikrishnan83** [comment]
@Rajaneeshkumar currently the release is planned for 30th December 2024 (Monday).

**Comment by harikrishnan83** [comment]
@Rajaneeshkumar We have just published [release 2.0.50](https://github.com/znsio/specmatic/releases/tag/2.0.50) ([Node module](https://pypi.org/project/specmatic/) is also updated). FYI. Thanks.
